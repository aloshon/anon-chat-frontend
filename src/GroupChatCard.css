
.mt-4.group-chat-card {
    margin-top: 10px;
    height: 300px;
}
.group-chat-card-header {
    margin: 5px;
    color: #666666;
    overflow: auto;
    max-height: 100px;
}

.group-chat-card-body {
    height: 140px;
    overflow: auto;
    margin-bottom: 0px;
    text-align: left;
}

.group-chat-card-footer {
    position: relative;
    bottom: 0;
}

@media (min-width: 900px) {
    .mt-4.group-chat-card {
        display:inline-block;
        width: 47%;
        vertical-align: top;
        margin-left: 10px;
    }
}

