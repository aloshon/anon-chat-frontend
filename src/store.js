import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);