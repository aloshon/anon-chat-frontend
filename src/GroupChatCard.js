import React, {useContext} from "react";
import Card from "react-bootstrap/Card";
import {NavLink} from "react-router-dom";
import Button from "react-bootstrap/Button";
import UserContext from "./UserContext";
import "./GroupChatCard.css";
import {useSelector} from "react-redux";

/**
 * GroupChatCard renders basic info on group chat and local timestamp.
 * If user is the creator then they are shown a button
 * to edit the group chat. There they can invite guests
 * or delete the group chat.
 */
const GroupChatCard = ({unique_id, title, description, timestamp, creator_id, displayInvitePage}) => {
    const {user} = useContext(UserContext);
    const darkMode = useSelector(state => state.darkMode);
    const localTime = new Date(timestamp);

    return (
        <Card className="mt-4 group-chat-card" style={{backgroundColor: darkMode.card}}>
            <NavLink exact to={`/chat/${unique_id}`}>
                <Card.Header 
                as="h3"
                className="group-chat-card-header"
                style={{color: darkMode.text}}>
                    {title}
                </Card.Header>
            </NavLink>
            <Card.Body className="group-chat-card-body">
                    <Card.Text as="h6">{description}</Card.Text>
                <div className="group-chat-card-description">
                    <small>{localTime.toLocaleString()}</small>
                    {user.id === creator_id && <>
                    <Button className="group-chat-card-button" variant="success" onClick={() => displayInvitePage(unique_id)}>Edit</Button>
                    </>}
                </div>
            </Card.Body>
        </Card>
    )
}
export default GroupChatCard;