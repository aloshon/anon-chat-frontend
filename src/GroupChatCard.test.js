import { render, screen } from '@testing-library/react';
import GroupChatCard from './GroupChatCard';
import { Provider } from "react-redux";
import { store } from "./store"
import {BrowserRouter} from 'react-router-dom';
import UserContext from "./UserContext";


describe("GroupChatCard should render basic group chat info", () => {
    const user = {
        id: 1,
        username: "user1",
        isAdmin: false,
        blockList: []
    }

    function displayInvitePage(unique_id){
        return history.push(`/invite/${unique_id}`)
    }

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <UserContext.Provider value={{user}}>
                <GroupChatCard
                unique_id="04ffa447-5237-476e-8v24-fb8n0nc801c7"
                title="test"
                description="testing"
                timestamp="2021-04-28T17:51:35.359Z"
                creator={1}
                displayInvitePage={displayInvitePage}
                />
            </UserContext.Provider>
        </BrowserRouter>
      </Provider>
    );
    test("renders group chat card with basic info", () => {
      const title = screen.getByText("test");
      const description = screen.getByText("testing");
      expect(title).toBeInTheDocument();
      expect(description).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });