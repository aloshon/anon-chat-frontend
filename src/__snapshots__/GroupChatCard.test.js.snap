// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupChatCard should render basic group chat info matches snapshot 1`] = `<DocumentFragment />`;
